src/Repository/ApplicationConfigRepository.php

```diff
@@ -129,4 +132,24 @@ protected function attachCriteriaToQueryBuilder(QueryBuilder $qb, $criteria, $or
 
         return $qb;
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function import(array $data, string $type = null, string $now = null): void
+    {
+        // ACHTUNG!
```

Comment:
It's complicated, but I think we should pull the existing data and check it here instead of clearing the table or maybe do an [on duplicate key update](https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html)?

Reply:
agreed. should be easier to do with entities.

Reply:
completely rewritten to use Entities, and added a reference map to it as well, so that imported entities won't have to be fetched again from DB.